GlympsApp/Glymps Main Controllers/SettingsVC.swift

```diff
@@ -269,6 +279,105 @@ class SettingsVC: UITableViewController {
         self.present(termsOfServiceVC, animated: true, completion: nil)
     }
     
+    func logAmplitudeSignoutEvent() {
+        API.User.observeCurrentUser { (user) in
+            var signOutEventProperties: [AnyHashable : Any] = [:]
+            signOutEventProperties.updateValue(user.email as Any, forKey: "Email")
+            signOutEventProperties.updateValue(user.age as Any, forKey: "Age")
+            signOutEventProperties.updateValue(user.profession as Any, forKey: "Profession")
+            signOutEventProperties.updateValue(user.company as Any, forKey: "Company")
+            signOutEventProperties.updateValue(user.name as Any, forKey: "Name")
+            signOutEventProperties.updateValue(user.gender as Any, forKey: "Gender")
+            signOutEventProperties.updateValue(user.id as Any, forKey: "User ID")
+            signOutEventProperties.updateValue(user.coins as Any, forKey: "Number of Glymps Coins")
+            signOutEventProperties.updateValue(user.isPremium as Any, forKey: "Subscription Status")
+            signOutEventProperties.updateValue(user.minAge as Any, forKey: "Minimum Preferred Age")
+            signOutEventProperties.updateValue(user.maxAge as Any, forKey: "Maximum Preferred Age")
+            signOutEventProperties.updateValue(user.preferedGender as Any, forKey: "Preferred Gender")
+            Amplitude.instance().logEvent("Sign Out", withEventProperties: signOutEventProperties)
```

Comment:
These values are already added as user properties arent they?

Reply:
Yes, for sign in events, but not for sign out events. Unfortunately, it looks like you have to attribute user properties to ever event if you want to have user metrics on them. Maybe we can just have a global variable called "userProperties" (current user attribute dict) and reuse them for all Amplitude events. What do you think?